domains/pallets/operator-rewards/src/lib.rs

```diff
@@ -53,6 +55,21 @@ mod pallet {
     #[pallet::getter(fn block_rewards)]
     pub(super) type BlockRewards<T: Config> = StorageValue<_, T::Balance, ValueQuery>;
 
+    /// The domain transaction byte fee
+    ///
+    /// NOTE: we are using `ValueQuery` for convenience, which means the transactions in the domain block #1
+    // are not charged for storage fees.
+    #[pallet::storage]
+    #[pallet::getter(fn domain_transaction_byte_fee)]
+    pub(super) type DomainTransactionByteFee<T: Config> = StorageValue<_, T::Balance, ValueQuery>;
+
+    /// The next domain transaction byte fee, it will take affect after the execution of the current
+    /// block to ensure the operator are using the same fee for both validating and executing the domain
+    /// transaction in the next block.
+    #[pallet::storage]
+    pub(super) type NextDomainTransactionByteFee<T: Config> =
```

Comment:
I understand why this storage is introduced but we do not need this if the transaction_byte_fee is persisted after state transition. This also makes the implementation much simpler I believe

Reply:
>if the transaction_byte_fee is persisted after state transition.

Do you mean the byte fee take effect immediately when executing the inherent extrinsic?